README.md

```diff
@@ -214,6 +214,48 @@ assert(Object.keys(@const { a: 1, b: 2 }) === @const ["a", "b"]);
 assert("a" in @const { a: 1, b: 2 });
 ```
 
+## Usage in {`Map`|`Set`|`WeakMap`}
+
+It is possible to use a const object or const array as a key in a `Map`, and as a value in a `Set`. When using a const object or const array in this way, key/value equality behaves as expected.
+
+It is not possible to use a const object or const array as a key in a `WeakMap`, because const objects and const arrays are not `Objects`, and have no lifetime. Attempting to set a value in a WeakMap using a const object
```

Comment:
Why would they have no lifetime? If i make a massive const object, and then it becomes collectible, why would we prohibit engines from doing so?

Reply:
More to the point, an object is any value for which `Object(x) === x`, so these are definitely objects.

Reply:
It is collectible by the engine but like a string: the lifetime of the const object will not be observable.

Reply:
why deviate from language precedent? it's an object.